### PR TITLE
 [pulsar-client] add BooleanSchema

### PR DESCRIPTION
**Pull request by ambition119**
### Motivation

 [pulsar-client]  not support boolean schema

### Modifications

add BooleanSchema.java, encode and decode

### Verifying this change

- [✅ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API: (yes / **no**)
  - The schema: (yes / **no** / don't know)
  - The default values of configurations: (yes / **no**)
  - The wire protocol: (yes / **no**)
  - The rest endpoints: (yes / **no**)
  - The admin cli options: (yes / **no**)
  - Anything that affects deployment: (yes / **no** / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)
  - If a feature is not applicable for documentation, explain why?
[Boolean] (https://github.com/apache/pulsar/blob/bc55bded83b773b2149c6fe4e3ef428673308027/pulsar-io/hbase/src/main/java/org/apache/pulsar/io/hbase/sink/HbaseGenericRecordSink.java#L79) just encode and decode
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Comment by ambition119** [comment]
run Integration Tests 

org.apache.pulsar.tests.integration.functions.PulsarFunctionsThreadTest.pulsar-test-suite

File: pulsar-client-api/src/main/java/org/apache/pulsar/common/schema/SchemaType.java [hunk 1]
```diff
@@ -27,6 +27,11 @@
      */
     NONE,
 
+    /**
+     * boolean schema defined
+     */
+    Boolean,
```
[the remark this answers is not in the document]
Comment (ambition119) [reply pulsar-client-api/src/main/java/org/apache/pulsar/common/schema/SchemaType.java]: ok

> maybe `BOOL` or `BOOLEAN`?

ok

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/BooleanSchema.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    @Override
+    public void validate(byte[] message) {
+        if (message.length != 1) {
+            throw new SchemaSerializationException("Size of data received by BooleanSchema is not 1");
+        }
+    }
+
+    @Override
+    public byte[] encode(Boolean message) {
+        return new byte[]{(byte)(message ? -1 : 0)};
```
[the remark this answers is not in the document]
Comment (ambition119) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/BooleanSchema.java]: > @ambition119 any reason why it is `-1`?

fix 1

File: pulsar-client-api/src/main/java/org/apache/pulsar/common/schema/SchemaType.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ambition119) [reply pulsar-client-api/src/main/java/org/apache/pulsar/common/schema/SchemaType.java]: > maybe `BOOL` or `BOOLEAN`?

ok

**Comment by ambition119** [comment]
rerun java tests

Pulsar SQL :: Pulsar Presto Distribution ........... FAILURE [ 18.788 s]

run Integration Tests

org.apache.pulsar.tests.integration.functions.PulsarFunctionsThreadTest.pulsar-test-suite

**Comment by ambition119** [comment]
rerun java tests

Pulsar SQL :: Pulsar Presto Distribution ........... FAILURE [ 18.788 s]

**Comment by ambition119** [comment]
run java8 tests

Pulsar SQL :: Pulsar Presto Distribution ........... FAILURE [ 18.788 s]

**Comment by ambition119** [comment]
run Integration Tests

[ERROR] tiered-storage-test-suite(org.apache.pulsar.tests.integration.offload.TestS3Offload)  Time elapsed: 11.732 s  <<< FAILURE!
java.lang.NoClassDefFoundError: org/apache/pulsar/client/admin/PulsarAdmin

**Comment by ambition119** [comment]
run Integration Tests

**Comment by sijie** [comment]
run integration tests

**Comment by ambition119** [comment]
run integration tests

org.apache.pulsar.tests.integration.functions.PulsarFunctionsThreadTest.pulsar-test-suite

**Comment by ambition119** [comment]
run Integration Tests

**Pull request merged**